airbyte-commons-worker/src/main/java/io/airbyte/workers/internal/DefaultAirbyteSource.java

```diff
@@ -78,8 +85,9 @@ public void start(final WorkerSourceConfig sourceConfig, final Path jobRoot) thr
         WorkerConstants.SOURCE_CONFIG_JSON_FILENAME,
         Jsons.serialize(sourceConfig.getSourceConnectionConfiguration()),
         WorkerConstants.SOURCE_CATALOG_JSON_FILENAME,
-        Jsons.serialize(sourceConfig.getCatalog()),
+        protocolSerializer.serialize(sourceConfig.getCatalog()),
         sourceConfig.getState() == null ? null : WorkerConstants.INPUT_STATE_JSON_FILENAME,
+        // TODO We should be passing a typed state here and use the protocolSerializer
```

Comment:
is this TODO still valid here?

Reply:
that's coming from https://github.com/airbytehq/airbyte/pull/20745/files#diff-4f22c30165c568984044a6200e904158f440f8d1b78cf2e588e99fe2561d4005R90

the relevant diff for this PR is https://github.com/airbytehq/airbyte/pull/19240/files/7adbe361e18e12c8850afad0f9e4dc0d8934c906..433c536a747f4c8f79dfae4a8cbf59211a5f7f42 (currently the base branch is set to https://github.com/airbytehq/airbyte/pull/20036, but I also took the commits from https://github.com/airbytehq/airbyte/pull/20745 for the sake of making everything compile)